troposphere/static/js/components/images/detail/availability/AvailabilityView.jsx

```diff
@@ -51,7 +52,35 @@ export default React.createClass({
 
     render() {
         let { version } = this.props;
+        let endDate = version.get('end_date'),
+            isEndDated = endDate && endDate.isValid();
+        // No availability if the version *OR* parent-image are end-dated.
+        if(!isEndDated) {
+            let image = version.get('image'),
+                end_date = moment(image.end_date);
+            isEndDated = end_date && end_date.isValid();
```

Comment:
This works, but I'm curious if we might want to move this into `models/Image` - so we could do this in the model's `parse`, and make the `moment` date-time available for all using this.